src/ceph.c

```diff
@@ -913,27 +914,34 @@ static int node_handler_fetch_data(void *arg, const char *val,
       // return after saving avgcount - don't dispatch value
       // until latency calculation
       return 0;
-    } else {
-      double sum, result;
-      sscanf(val, "%lf", &sum);
-
+    } else if (vtmp->avgcount_exists == 0) {
```

Comment:
As I can see - ```ds_name``` has a long metric name, not just ```avgcount``` or ```sum```. So, I replaced ```avgcount_exists``` with readable enum.

Reply:
That's still a state machine and as a result extremely brittle. It's sensitive to new metrics being added and depends on the order of metrics.

Can't you use the existing `has_suffix()` function to determine which metric it is?

Reply:
I tried to find easy way to do it and did some tests.
Unfortunately, ```key``` and ```ds_name``` doesn't contain suffix we need. ```ceph_cb_number``` cut suffix for latency metrics. Do you think it should be fixed within this pull request?

Reply:
Yes, I think this should be fixed in this PR: the reason we have to revisit this code for a new Ceph release it because it is so brittle. A good fix for this problem should ensure that the code works even if Ceph should decide to

*   add a median, 95th percentile or distribution of the latency, or
*   re-order the fields, e.g. "sum" first, "avgcount" second.